### PR TITLE
Compose upgrade

### PR DESCRIPTION
**Pull request by bmarty**
Replacement for #1867 and do some cleanup on the codebase.

### CODE REVIEW

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootNode.kt [hunk 1]
```diff
@@ -72,8 +70,8 @@ class CreateRoomRootNode @AssistedInject constructor(
             state = state,
             modifier = modifier,
             onClosePressed = this::navigateUp,
-            onNewRoomClicked = callback::onCreateNewRoom,
-            onOpenDM = callback::onStartChatSuccess,
+            onNewRoomClicked = ::onCreateNewRoom,
+            onOpenDM = ::onStartChatSuccess,
```
Comment (bmarty) [inline features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootNode.kt:74]: The compose compiler (or Kotlin compiler?) did not like the previous version

Comment (jmartinesp) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootNode.kt:74]: We could just use `= { callback.onCreateNewRoom() }` there, but this is fine too.

**Review by jmartinesp — APPROVED** [review]
LGTM, thanks!

File: features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootNode.kt (hunk 1 above)
Comment (bmarty) [reply features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/root/CreateRoomRootNode.kt:74]: Yes, but I prefer to keep the same pattern everywhere.

**Comment by bmarty** [comment]
Failing test is a flaky test that need to be fixed in a dedicated PR.

**Pull request merged**